### DISCUSSION TITLE
Confusion about branch protections

### DISCUSSION THREAD

**Discussion by omerlh** [renovatebot/renovate#25302] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.109.4 (helm chart version)

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

All PRs followed our information security policies - enforced by branch protection rules - e.g. tests are passing and at least one reviewer

### What you tried so far.

I am not sure where to look even. We had at least 2 cases where renovate merged a PR with test passing but no review by one of our team members. It is possible because we allowed the bot user renovate is using to bypass branch protection for a different case - but we don't want renovate to do that.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
So you want Renovate to bypass branch protection but also don't want it to bypass branch protection? I'm not sure we can help you with that

**Comment by omerlh** [comment]
No, I never want to renovate to bypass branch protection. We have another github action running on main that build docker image and commit image tag directly to main. This is the only case we consider as valid for bypassing branch protection. But github users costs money so I am reusing the users here...

**Reply by rarkins** [reply]
You either need to give Renovate an account that can't bypass branch protections, or you need to remove automerge from Renovate settings. GitHub's branch protections capabilities change all the time, and they're not always easy to determine via the API either. We are not going to try to reverse engineer and check every possible setting GitHub has - which is a moving target - so that we can make sure we don't bypass any. Instead, apply Branch protections to your Renovate user so that it's fundamentally impossible for it to bypass them.

Regarding user cost, GitHub Apps don't cost money to add, so you try that instead of a "real" user account.
